From 5c3bb1839cac52828756f9ddb98f49d586853991 Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Wed, 16 Nov 2011 17:38:45 -0500
Subject: [PATCH 1/4] Revert "iconview: layout items immediately when setting
 a GtkTreeModel"

It makes GtkIconView segfault on GTK 2.24.

gtk_icon_view_expose() calls gtk_icon_view_layout() first thing if
there's a layout queued anyway, so we wouldn't end up in the same
situation causing the crash the original patch is supposed to fix.

This reverts commit 5a03f4a6a50237d86959f596dda143dfc2f040d1.

https://bugzilla.gnome.org/show_bug.cgi?id=663138
---
 gtk/gtkiconview.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/gtk/gtkiconview.c b/gtk/gtkiconview.c
index 62877fa..54b3f3b 100644
--- a/gtk/gtkiconview.c
+++ b/gtk/gtkiconview.c
@@ -5423,7 +5423,7 @@ gtk_icon_view_set_model (GtkIconView *icon_view,
 
       gtk_icon_view_build_items (icon_view);
 
-      gtk_icon_view_layout (icon_view);
+      gtk_icon_view_queue_layout (icon_view);
     }
 
   g_object_notify (G_OBJECT (icon_view), "model");  
-- 
1.7.7.1

